Misc docs updates (#16122)
unoplatform/uno: doc/articles/migrating-to-single-project.md

## What is the Uno.Sdk

The Uno.Sdk is used to abstract the complexity needed to build cross-platform projects, while still ensuring that you have the ability to customize the configuration to your needs.

This means that values such as the `SupportedOSPlatformVersion`, which have been in the template's `Directory.Build.props` in previous Uno Platform templates, now have a default value set to the minimum supported by Uno Platform.

If you need to be more restrictive for your specific project you can provide your overrides for the platform(s) that you need.

## Project Structure

In the Uno Platform Single Project, all platform-specific code is put together in one project. Instead of having separate projects for each platform, everything is organized within a `Platforms` folder. This folder contains sub-folders for each platform your project targets, like Android, iOS, Skia, WebAssembly, and Windows. Each platform folder stores the code and resources needed for that platform, making development easier to manage for multi-platform projects.

## Upgrading an existing project to Single Project

> The following sections detail how to upgrade an existing project to use a single project with the Uno.Sdk. Those modifications are made in-place and **it is strongly recommended to work on a source-controlled environment**.

Reviewer: ```suggestion
The following sections detail how to upgrade an existing project to use a single project with the Uno.Sdk. Those modifications are made in place and **it is strongly recommended to work on a source-controlled environment**.
```

While following this guide, you can compare with a new empty project created using [our Wizard](xref:Uno.GetStarted.Wizard), or using [dotnet new templates](xref:Uno.GetStarted.dotnet-new).

### Migrating to a Single Project

> Migrating from a 5.1 or earlier template is a matter merging all the csproj files for all the platforms heads down to one csproj file.

Reviewer: ```suggestion
Migrating from a 5.1 or earlier template is a matter of merging all the csproj files for all the platforms heads down to one csproj file.
```

1. Next to the solution, create a `global.json` file
    ```json
    {
        "msbuild-sdks": {
            "Uno.Sdk": "{Current Version of Uno.WinUI/Uno.Sdk}"
        }
    }
    ```
1. From the empty app created at step one:
    - Take the contents of the `MyApp.csproj` and copy it over your own `MyApp.csproj`
    - Take the folder named `Properties` and copy it over your own `MyApp`
1. If you have your own `PackageReference` and `ProjectReference`, make sure to preserve them.
1. From the empty app created at step one, take the `Platforms` folder and copy it over your own `MyApp` folder.
1. In you app's `MyApp.csproj`, create the following block:
   ```xml
   <Choose>
     <When Condition="'$(TargetFramework)'=='net8.0-ios'">
     </When>
     <When Condition="'$(TargetFramework)'=='net8.0-android'">
     </When>
     <When Condition="'$(TargetFramework)'=='net8.0-maccatalyst'">
     </When>
     <When Condition="'$(TargetFramework)'=='net8.0-windows10.0.19041'">
     </When>
     <When Condition="'$(TargetFramework)'=='net8.0-desktop'">
     </When>
     <When Condition="'$(TargetFramework)'=='net8.0-browserwasm'">
     </When>
   </Choose>
   ```

>    This will allow to place custom project configuration in each of those sections. Use the ones that are relevant to your project targets.
> 1. From the empty app created at step one, take the `App.xaml` and `App.xaml.cs` and copy those over your own `MyApp` folder.
> 1. Copy the individual modifications you made to your `AppResources.xaml` and `App.cs` into `App.xaml` and `App.xaml.cs`, respectively. Those can for instance be changes made to `OnLaunched`, `InitializeLogging` or `RegisterRoutes`.

Reviewer: ```suggestion
   This will allow us to place custom project configurations in each of those sections. Use the ones that are relevant to your project targets.
1. From the empty app created at step one, take the `App.xaml` and `App.xaml.cs` and copy those over your own `MyApp` folder.
1. Copy the individual modifications you made to your `AppResources.xaml` and `App.cs` into `App.xaml` and `App.xaml.cs`, respectively. Those can, for instance, be changes made to `OnLaunched`, `InitializeLogging`, or `RegisterRoutes`.
```

    1. If you've made modifications to the `base.props` file, you can copy those over to the `MyApp.csproj`, outside the `Choose` section.
    1. If you've made individual changes to the `AppHead.xaml` file, copy those over to the `App.xaml` file
    1. If you've made individual changes to the `AppHead.xaml.cs` file, copy those over to the `App.xaml.cs` file
    1. Delete the `MyApp.[Shared|Base]` folder
1. In All the `.cs` files, replace `AppHead` with `App`
1. Once done, you can remove the `Platforms` folder from your solution explorer, as all other platform heads have been deleted.
1. If you have a `MyApp.Server` project, make sure to change from:
    ```xml
    <ProjectReference Include="..\MyApp.Wasm\MyApp.Wasm.csproj" />
    ```
    to
    ```xml
    <ProjectReference 
        Include="..\MyApp\MyApp.csproj"
        SetTargetFramework="TargetFramework=net8.0-browserwasm"
        ReferenceOutputAssembly="false" />
    ```
1. From the empty app created at step one, take the `Directory.Build.props` and `Directory.Build.targets` files and copy those over your own root folder. Make sure to copy individual changes made in your file, such as `CSharpVersion` or `Nullable`.

> 1. From the empty app created at step one, take the `Directory.Packages.props`file and copy it over your own root folder. All of the nuget packages can be removed from this file, as the Uno.SDK provides those automatically.

Reviewer: ```suggestion
1. From the empty app created at step one, take the `Directory.Packages.props`file and copy it over to your own root folder. All of the Nuget packages can be removed from this file, as the Uno.SDK provides those automatically.
```
